### PR TITLE
Implements clickatell sms provider

### LINKED ISSUE #604: Add Clickatell SMS provider
### What?
New [Clickatell](https://github.com/clickatell/clickatell-node) SMS Provider.

### Why?
To have the ability to use Clickatell as a provider.

### How?
Add new provider based on [this guide](https://docs.notifire.co/docs/community/create-provider)

### Usage
Example for Clickatell SMS provider registration needs to be finalized. 

```ts
  new ClickatellSmsProvider({
    // API is missing
  })

```

### PR DESCRIPTION
**Pull request by Abobos**
- **What kind of change does this PR introduce?** 
 Feature (closes #604)

- **What is the current behavior?** (You can also link to an open issue here)
Currently, we don't implement Clickatell provider for the SMS channel

- **What is the new behavior (if this is a feature change)?**
Avail Clickatell SMS Provider

### CODE REVIEW

File: providers/clickatell/test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ClickatellSmsProvider } from './src/index';
+
+const provider = new ClickatellSmsProvider({
+  apiKey: 'otbjTKxDRlGxGlA6wiSCKQ==',
```
Comment (jainpawan21) [inline providers/clickatell/test.ts:4]: I think you exposed your key by mistake

Comment (Abobos) [reply providers/clickatell/test.ts]: @jainpawan21 this has been removed. It was a file I used for testing the functionality

**Comment by florian-lefebvre** [comment]
Hey @Abobos @jainpawan21 what's preventing this PR from being merged?

**Comment by Abobos** [comment]
> Hey @Abobos @jainpawan21 what's preventing this PR from being merged?

It needs to be reviewed cc : @scopsy

**Review by scopsy — APPROVED** [review]
Thank you @Abobos for your contribution and sorry it took so long to merge ❤️

**Comment by Abobos** [comment]
> Thank you @Abobos for your contribution and sorry it took so long to merge ❤️

@scopsy no worries..Looking forward to contribute more

**Pull request merged**